src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "lora_helper.hpp"
+#include "image_generation/models/unet_inference.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace genai {
+
+// Static Batch-Size 1 variant of UNetInference
+class UNet2DConditionModel::UNetInferenceStaticBS1 : public UNet2DConditionModel::UNetInference {
+public:
+    virtual void compile(std::shared_ptr<ov::Model> model,
+                         const std::string& device,
+                         const ov::AnyMap& properties) override {
+
+        // All shapes for input/output tensors should be static. 
+        // Double check this and throw runtime error if it's not the case.
+        for (auto& input : model->inputs()) {
+            if (input.get_partial_shape().is_dynamic()) {
+                throw std::runtime_error(
+                    "UNetInferenceStaticBS1::compile: input tensor " + input.get_any_name() +
+                    " shape is dynamic. Tensors must be reshaped to be static before compile is invoked.");
```

Comment:
you can use `OPENVINO_ASSERT(!input.get_partial_shape().is_dynamic(), "erro text")`

Reply:
addressed in this PR: https://github.com/openvinotoolkit/openvino.genai/pull/1172